src/app/shared/navigation/navigation.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable, NgZone } from '@angular/core';
+import { Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NavigationService {
+
+  constructor(private ngZone: NgZone, private router: Router) { }
+
+  /*
+    Using this to replace routerLink where necessary as I was getting Angular Zone related/delayed loading warnings:
+    Console -> "Navigation triggered outside Angular zone, did you forget to call 'ngZone.run()'?"
+  */
+  public navigate(subpaths: any[]): void {
+    // needed in order to trigger appropriate change detection when user wants to change navigation
+    // became necessary after implementing "getTripImgUrl()" in TripService in relation with Trip Resolvers
+    this.ngZone.run(() => this.router.navigate(subpaths)).then();
```

Comment:
It's an interesting one. Not sure if you've viewed this page on change detection strategies. One of them certainly is ngZone.run https://angular.io/guide/change-detection

Reply:
This global use of ngZone is not normal and is an indication of a different problem. Typically, you'll need to use ngZone if you're using JS based libraries but trying to do Angular things in the callbacks of those libraries, for example.

I'll accept this stuff but it should not be the default solution. You should look for why this might be happening. My bet would be the use of the Google places library.

Reply:
Yea, I was struggling with this for a while. It's definitely the Google Places library. Google APIs are cool and all but it doesn't mix as well with Angular as I originally anticipated. I'll have to look further into this and see if I can get this to work better.